### PR TITLE
feat(usersettings): add separate setting for streaming region

### PR DESCRIPTION
**Pull request by gauthier-th**
#### Description

Currently, the "Currently Streaming On" information is based on the Discover Region setting. This PR adds a new setting to specify which region should be used to display the streaming region.

#### Screenshot (if UI-related)

New setting in general settings as well as in user settings:
image

#### To-Dos

- [x] Successful build `pnpm build`
- [x] Translation keys `pnpm i18n:extract`
- [x] Database migration (if required)

#### Issues Fixed or Closed

- Re #890

### CODE REVIEW

File: server/lib/settings/migrations/0004_migrate_region_setting.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { AllSettings } from '@server/lib/settings';
+
+const migrateHostname = (settings: any): AllSettings => {
```
Comment (fallenbagel) [inline server/lib/settings/migrations/0004_migrate_region_setting.ts]: Why is this called `migrateHostname`?

**Review by Fallenbagel — CHANGES_REQUESTED** [review]

File: server/lib/settings/migrations/0004_migrate_region_setting.ts (hunk 1 above)
Comment (gauthier-th) [reply server/lib/settings/migrations/0004_migrate_region_setting.ts]: Copy/paste error, let me fix it

**Review by Fallenbagel — APPROVED** [review]
LGTM

**Pull request merged**